### PR TITLE
CP-15132: Systemd conversion

### PR DESCRIPTION
**Pull request by rafalmiel**
Notify systemd when the daemon is ready

Signed-off-by: Rafal Mielniczuk <rafal.mielniczuk@citrix.com>

### CODE REVIEW

File: networkd/networkd.ml [hunk 1]
```diff
@@ -108,6 +110,8 @@ let _ =
 		Debug.with_thread_associated "main" start server
 	) ();
 
+        ignore (Daemon.notify Daemon.State.Ready);
```
Comment (robhoes) [inline networkd/networkd.ml]: Whitespace...

**Comment by robhoes** [comment]
:shipit: apart from the minor issue. This one definitely needs "notify" for the moment.

**Comment by robhoes** [comment]
Why is the CA-201728 patch included here?

File: networkd/networkd.ml [hunk 2]
```diff
@@ -80,6 +80,8 @@ let doc = String.concat "\n" [
 	"This service looks after host network configuration, including setting up bridges and/or openvswitch instances, configuring IP addresses etc.";
 ]
 
+exception Notify_failed
```
Comment (robhoes) [inline networkd/networkd.ml]: Why is this exception defined here? It's not there in the other daemons.

Comment (rafalmiel) [inline networkd/networkd.ml]: It is fix to one of the problems that arised after changes. I will remove the exception. It's not needed indeed

**Pull request merged**